js/app.js

```diff
@@ -0,0 +1,454 @@
+    winner = false
+  } else if (cell.value > 0){
+    cell.innerText = ""
+    revealed[y][x] = 1
+    
+  } else if (cell.value === 0) {
+    cell.innerText = ""
+    cascade(x, y)
+  }
+  render()
+
+  }
+  
+
+
+
+
+
+
+
+function createNeighborArr(x, y) {
+  
+  return [checkNW(x, y), checkN(x, y),checkNE(x, y),checkW(x, y),checkE(x, y), checkSW(x, y), checkS(x, y), checkSE(x, y)]
+}
+
+
+
+
+function checkNW(x, y) {
+  if (y > 0 && x > 0) {
+    return {
+      name: 'northWest',
+      value: gameBoard[y-1][x-1],
+      yLocation: y-1,
+      xLocation: x-1,
+      status: null
+    }
+  } else {return {
+    name: 'northWest',
+    value: null,
+    yLocation: null,
+    xLocation: null,
+    status: null
+  }}
+}
+function checkN(x, y) {
+ 
+  if (y > 0) {
+    return {
+      name: 'north',
+      value: gameBoard[y-1][x],
+      yLocation: y-1,
+      xLocation: x,
+      status: null
+    }
+  } else {return {
+    name: 'north',
+    value: null,
+    yLocation: null,
+    xLocation: null,
+    status: null
+  }}
+}
+function checkNE(x, y) {
+ 
+  if (y > 0 && x<9) {
+    return {
+      name: 'northEast',
+      value: gameBoard[y-1][x+1],
+      yLocation: y-1,
+      xLocation: x+1,
+      status: null
+    }
+  } else {return {
+    name: 'northEast',
+    value: null,
+    yLocation: null,
+    xLocation: null,
+    status: null
+  }}
+}
+function checkW(x, y) {
+  
+  if (x>0) {
+    return {
+      name: 'west',
+      value: gameBoard[y][x-1],
+      yLocation: y,
+      xLocation: x-1,
+      status: null
+    }
+  } else {return {
+    name: 'west',
+    value: null,
+    yLocation: null,
+    xLocation: null,
+    status: null
+  }}
+}
+function checkE(x, y) {
+  
+  if (x<9) {
+    return {
+      name: 'east',
+      value: gameBoard[y][x+1],
+      yLocation: y,
+      xLocation: x+1,
+      status: null
+    }
+  } else {return {
+    name: 'east',
+    value: null,
+    yLocation: null,
+    xLocation: null,
+    status: null
+  }}
+}
+function checkSW(x, y) {
+  
+  if (y < 7 && x > 0) {
+    return {
+      name: 'southWest',
+      value: gameBoard[y+1][x-1],
+      yLocation: y+1,
+      xLocation: x-1,
+      status: null
+    }
+  } else {return {
+    name: 'southWest',
+    value: null,
+    yLocation: null,
+    xLocation: null,
+    status: null
+  }}
+}
+function checkS(x, y) {
+
+  if (y < 7) {
+    return {
+      name: 'south',
+      value: gameBoard[y+1][x],
+      yLocation: y+1,
+      xLocation: x,
+      status: null
+    }
+  } else {return {
+    name: 'south',
+    value: null,
+    yLocation: null,
+    xLocation: null,
+    status: null
+  }}
+}
+function checkSE(x, y) {
+
+  if (y < 7 && x < 7) {
+    return {
+      name: 'southEast',
+      value: gameBoard[y+1][x+1],
+      yLocation: y+1,
+      xLocation: x+1,
+      status: null
+    }
+  } else {return {
+    name: 'southEast',
+    value: null,
+    yLocation: null,
+    xLocation: null,
+    status: null
+  }}
+}
```

Comment:
The objects here are great, but you don't need `status` or `name` properties because you never use them in the rest of your code. Also in all of your `else`s you are returning essentially the same object with everything in it set as `null` - you could just build a `const` with the values set as null and return that instead. For example:

```javascript

const nullCell = {
  value: null,
  yLocation: null,
  xLocation: null,
}

function checkS(x, y) {
  if (y < 7) {
    return {
      value: gameBoard[y+1][x],
      yLocation: y+1,
      xLocation: x,
    }
  } else {
     return nullCell
  }
}

function checkSE(x, y) {
  if (y < 7 && x < 7) {
    return {
      value: gameBoard[y+1][x+1],
      yLocation: y+1,
      xLocation: x+1,
    }
  } else {
    return nullCell
  }
}

```